libraries/dotnet/EnergyOriginEventStore/EnergyOriginEventStore/EventStore/Memory/MemoryEventConsumer.cs

```diff
@@ -0,0 +1,40 @@
+using EnergyOriginEventStore.EventStore.Internal;
+using EnergyOriginEventStore.EventStore.Serialization;
+
+namespace EnergyOriginEventStore.EventStore.Memory;
+
+internal class MemoryEventConsumer : IEventConsumer
+{
+    private readonly MemoryEventStore _store;
+    private readonly IUnpacker _unpacker;
+    private readonly Dictionary<Type, IEnumerable<Action<Event<EventModel>>>> _handlers;
+    private readonly MemoryPointer? _pointer;
+    private readonly string _topicPrefix;
+
+    public MemoryEventConsumer(IUnpacker unpacker, Dictionary<Type, IEnumerable<Action<Event<EventModel>>>> handlers, MemoryEventStore store, string topicPrefix, MemoryPointer? pointer)
+    {
+        _store = store;
+        _unpacker = unpacker;
+        _handlers = handlers;
+        _topicPrefix = topicPrefix;
+        _pointer = pointer;
+
+        Task.Run(async () => await _store.Attach(this).ConfigureAwait(false));
```

Comment:
Generally the constructor should be fast and not perform actual work. Furthermore, since it is not possible to have an async constructor, there is a need to do Task.Run().
But if we create a static method called e.g. `Create` we can let that method be async and then call `.Attach()` on MemoryEventStore.

Reply:
We are going to leave this as is and see what changes we potentially need in the interface. We may want to have the `build()` method be async.